### PR TITLE
Code QL pre-migration tasks: update articles relating to CodeQL for JavaScript

### PR DESCRIPTION
**Pull request by mchammer01**
See https://github.com/github/product-documentation/issues/2165 for details.
The first commit changes titles.

### CODE REVIEW

**Comment by mchammer01** [comment]
2nd commit adds some intros but I'm afraid I won't get much further before we have to leave for the mini-summit.

**Comment by mchammer01** [comment]
Please note that I haven't made any changes to do with content models and localization checklists.

File: docs/language/learn-ql/javascript/introduce-libraries-js.rst [hunk 1]
```diff
@@ -1,5 +1,7 @@
-Introducing the CodeQL libraries for JavaScript
-===============================================
+CodeQL libraries for JavaScript
+===============================
+
+This article is an overview of the extensive libraries you can use to analyze databases generated from JavaScript codebases. Using thes libraries makes it easier for you to write queries.
```
Comment (intrigus-lgtm) [inline docs/language/learn-ql/javascript/introduce-libraries-js.rst]: Typo: "thes libraries"

File: docs/language/learn-ql/javascript/introduce-libraries-ts.rst [hunk 2]
```diff
@@ -1,5 +1,7 @@
-Introducing the CodeQL libraries for TypeScript
-===============================================
+CodeQL libraries for TypeScript
+===============================
+
+This article is an overview of the libraries you can use to analyze databases generated from TypeScript codebases. Using thes libraries makes it easier for you to write queries.
```
Comment (intrigus-lgtm) [inline docs/language/learn-ql/javascript/introduce-libraries-ts.rst]: Same typo here.

File: docs/language/learn-ql/javascript/type-tracking.rst [hunk 3]
```diff
@@ -1,5 +1,5 @@
-Tutorial: API modelling using type tracking
-===========================================
+Using type tracking for API modeling
```
Comment (intrigus-lgtm) [inline docs/language/learn-ql/javascript/type-tracking.rst]: Nitpicking: If the docs don't use American English, this should probably be `modelling` with two `l`s?

**Comment by mchammer01** [comment]
Thanks for the valuable review @intrigus-lgtm 👍 
This PR has been raised as a draft so I wasn't expecting reviews just yet 😉 
It's just a starting point and I have a lot more work to do but I'll ensure I'll take your comments (especially the ones about the typos 😱 ) into consideration.
Our documentation is meant to be written in US English but I know that we need to rationalise spellings and check that this is true throughout the whole set of articles.
Thanks again for your help!

**Comment by mchammer01** [comment]
This PR is now ready for review.
The work in this PR contains pre-migration tasks for the CodeQL documentation. This includes:
- updating article titles
- adding intros to each articles
- updating links as per the new article title

**Comment by mchammer01** [comment]
Apologies for the automatic review request @jf205 and @shati-patel but I can't seem to be able to remove that (is this a permissions thing on this repo)?

**Comment by felicitymay** [comment]
Since this is small text changes, I'm removing @jf205 ad @shati-patel as reviewers. Anyone in the docs team can review this PR.

File: docs/language/learn-ql/javascript/introduce-libraries-ts.rst [hunk 4]
```diff
@@ -10,7 +12,7 @@ Support for analyzing TypeScript code is bundled with the CodeQL libraries for J
 
    import javascript
 
-The :doc:`CodeQL library introduction for JavaScript <introduce-libraries-js>` covers most of this library, and is also relevant for TypeScript analysis. This document supplements the JavaScript documentation with the TypeScript-specific classes and predicates.
+The :doc:`CodeQL libraries for JavaScript <introduce-libraries-js>` covers most of this library, and is also relevant for TypeScript analysis. This document supplements the JavaScript documentation with the TypeScript-specific classes and predicates.
```
Comment (felicitymay) [inline docs/language/learn-ql/javascript/introduce-libraries-ts.rst]: With the change in topic name, it reads a little weirdly now. Suggest dropping "the".

```suggestion
:doc:`CodeQL libraries for JavaScript <introduce-libraries-js>` covers most of this library, and is also relevant for TypeScript analysis. This document supplements the JavaScript documentation with the TypeScript-specific classes and predicates.
```

File: docs/language/learn-ql/javascript/type-tracking.rst (hunk 3 above)
Comment (felicitymay) [reply docs/language/learn-ql/javascript/type-tracking.rst]: Thanks for highlighting this. The docs aim to use American English, so this is correct.

File: docs/language/learn-ql/javascript/type-tracking.rst [hunk 5]
```diff
@@ -489,7 +489,7 @@ Prefer type tracking when:
 Prefer data-flow configurations when:
 
 - Tracking user-controlled data -- use `taint tracking <https://help.semmle.com/QL/learn-ql/javascript/dataflow.html#using-global-taint-tracking>`__.
-- Differentiating between different kinds of user-controlled data -- use :doc:`flow labels <flow-labels>`.
+- Differentiating between different kinds of user-controlled data -- use :doc:`Using flow labels for precise data flow analysis <flow-labels>`.
```
Comment (felicitymay) [inline docs/language/learn-ql/javascript/type-tracking.rst]: With the link name change, suggest:
```suggestion
- Differentiating between different kinds of user-controlled data -- see :doc:`Using flow labels for precise data flow analysis <flow-labels>`.
```

**Comment by mchammer01** [comment]
@felicitymay - thanks for your review. I believe I have addressed your comments.
I am keen to get this merged today, as discussed in our team meeting. 
We can still iterate if there are further things to tweak or rationalize between languages.

File: docs/language/learn-ql/javascript/type-tracking.rst [hunk 6]
```diff
@@ -1,9 +1,11 @@
-Tutorial: API modelling using type tracking
-===========================================
+Using type tracking for API modeling
+====================================
 
-This tutorial demonstrates how to build a simple model of the Firebase API
+You can build a simple model of the Firebase API
```
Comment (felicitymay) [inline docs/language/learn-ql/javascript/type-tracking.rst]: It looks as if Firebase is just used as an example, so suggest making this a little more general:

```suggestion
You can track data through an API by creating a model 
```

File: docs/language/learn-ql/javascript/dataflow-cheat-sheet.rst [hunk 7]
```diff
@@ -1,5 +1,5 @@
-Data flow cheat sheet
-=====================
+Data flow cheat sheet for JavaScript
+====================================
 
 This page describes parts of the JavaScript libraries commonly used for variant analysis and in data flow queries.
```
Comment (felicitymay) [inline docs/language/learn-ql/javascript/dataflow-cheat-sheet.rst]:
```suggestion
This article describes parts of the JavaScript libraries commonly used for variant analysis and in data flow queries.
```

**Comment by mchammer01** [comment]
@felicitymay - I have accepted your suggestions. Can you double-check you are ok with things now? Thanks.

**Comment by felicitymay** [comment]
All LGTM 💚

**Pull request merged**